integration/chartconfig/chartconfig.go

```diff
@@ -4,18 +4,128 @@ package chartconfig
 
 import (
 	"bytes"
+	"context"
+	"fmt"
 	"html/template"
+	"log"
+	"os"
+	"strings"
 
+	"github.com/giantswarm/apprclient"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
 	"github.com/giantswarm/e2etemplates/pkg/e2etemplates"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+	"k8s.io/helm/pkg/helm"
+
+	"github.com/giantswarm/chart-operator/integration/env"
+	"github.com/giantswarm/chart-operator/integration/templates"
 )
 
-func ExecuteChartValuesTemplate(ccv e2etemplates.ApiextensionsChartConfigValues) (string, error) {
+func DeleteResources(ctx context.Context, helmClient *helmclient.Client, l micrologger.Logger) error {
+	items := []string{"cnr-server", "apiextensions-chart-config-e2e"}
+
+	for _, item := range items {
+		err := helmClient.DeleteRelease(ctx, item, helm.DeletePurge(true))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	return nil
+}
+
+func ExecuteValuesTemplate(ccv e2etemplates.ApiextensionsChartConfigValues) (string, error) {
 	buf := &bytes.Buffer{}
 	chartValuesTemplate := template.Must(template.New("chartConfigChartValues").Parse(e2etemplates.ApiextensionsChartConfigE2EChartValues))
 	err := chartValuesTemplate.Execute(buf, ccv)
 	if err != nil {
-		return "", err
+		return "", microerror.Mask(err)
 	}
 
 	return buf.String(), nil
 }
+
+func InstallResources(ctx context.Context, h *framework.Host, helmClient *helmclient.Client, l micrologger.Logger) error {
+	err := initializeCNR(ctx, h, helmClient, l)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	version := fmt.Sprintf(":%s", env.CircleSHA())
+	err = h.InstallOperator("chart-operator", "chartconfig", templates.ChartOperatorValues, version)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func VersionBundleVersion(githubToken, testedVersion string) (string, error) {
+	if githubToken == "" {
+		return "", microerror.Maskf(invalidConfigError, "env var %#q must not be empty", env.EnvVarGithubBotToken)
+	}
+	if testedVersion == "" {
+		return "", microerror.Maskf(invalidConfigError, "env var %#q must not be empty", env.EnvVarTestedVersion)
```

Comment:
Let's please use these error types within constructors. Here the catch all would be `failedExecutionError`.

Reply:
Good catch. Done.